### ISSUE TITLE
[Server] [Discovery] dutyfree backend

### ISSUE THREAD

**Issue by YoussB** [concourse/dutyfree#119]
### Background:
- As the wise man says, for every frontend there is a backend to serve its needs.
- This is discovery into how will the backend of dutyfree look like. The current ideas that I have on mind are:
  - API backend with a small db (might be an overkill)
  - API backend with a yml file (might not be the best architecture)
  - Same as the current way: rendering static pages based on a yaml file.

### Acceptance Criteria:
- [ ] we should define the way the backend will look like.

**Comment by jomsie** [comment]
so had a bit of a discussion with @YoussB earlier, and we talked about a couple of things regarding the backend discovery:

- first it just needs to serve up resource names, versions, and descriptions (IIRC)
- when it comes to figuring out the contributor workflow, ie how to add a resource, how we check whether a resource is acceptable or not, how that resource gets updated when a new version comes out, etc etc, will be decided somewhat later, but we should probably keep those questions in the back of our mind when we are deciding on how we want our backend to be set up
- i think that having the backend with a db is the easiest and most futureproof way of doing things, since the most we'll have to do data wise is a migration, _BUT_ even that will affect the contributor workflow (i think)
- so long story short, we're still discussing 😅

**Comment by YoussB** [comment]
- working out of the branch `go-backdend`
- the current plan is to have a Golang api in the backend, but still unsure about the persistence layer.
- we're building a working skeleton for the api.

**Comment by YoussB** [comment]
- We now have our basic backend serving api and static files, this replaces the need to have a separate way to serve static files.
- The current code is experimental, but sets a base for the backend.
